addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1364,73 +1472,96 @@ var VideoWidget = MediaWidget.extend({
-                    });
-                    break;
+        const $textarea = this.$('textarea#o_video_text');
+        const code = $textarea.val().trim();
+        const uploadedVideo = this.$('.o_we_attachment_selected');
+        let query = null;
+        if (uploadedVideo.length && this.selectedAttachments.length) {
+            this._showUpload();
+            query = await this._createVideoNode(this.selectedAttachments[0].src, {
+                'uploaded': true,
+                'autoplay': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
+                'muted': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
+                'controls': this.isForBgVideo || this.$('input#o_video_hide_controls').is(':checked'),
+                'loop': this.isForBgVideo || this.$('input#o_video_loop').is(':checked')
+            });
+        } else if (code) {
+            if (code.includes("/web/image/")) {
+                for (const attachment of this.attachments) {
+                    if (code.includes(attachment["src"])) {
+                        await this._selectAttachement(attachment, false);
+                    }
+                }
+                if (!(this.selectedAttachments.length)) {
+                    // /web/image URL provided but no such resource in attachments
+                    this.$('#o_video_text').val("");
+                    await this._updateVideo();
+                }
+            } else {
+                this._showURL();
+                this.$(".o_video_user_upload").hide();
+                // Detect if we have an embed code rather than an URL
+                const embedMatch = code.match(/(src|href)=["']?([^"']+)?/);
+                if (embedMatch && embedMatch[2].length > 0 && embedMatch[2].indexOf('instagram')) {
+                    embedMatch[1] = embedMatch[2]; // Instagram embed code is different
+                }
+                const url = embedMatch ? embedMatch[1] : code;
+
+                query = await this._createVideoNode(url, {
+                    'autoplay': this.isForBgVideo || this.$('input#o_video_autoplay').is(':checked'),
+                    'hide_controls': this.isForBgVideo || this.$('input#o_video_hide_controls').is(':checked'),
+                    'loop': this.isForBgVideo || this.$('input#o_video_loop').is(':checked'),
+                    'hide_fullscreen': this.isForBgVideo || this.$('input#o_video_hide_fullscreen').is(':checked'),
+                    'hide_yt_logo': this.isForBgVideo || this.$('input#o_video_hide_yt_logo').is(':checked'),
+                    'hide_dm_logo': this.isForBgVideo || this.$('input#o_video_hide_dm_logo').is(':checked'),
+                    'hide_dm_share': this.isForBgVideo || this.$('input#o_video_hide_dm_share').is(':checked'),
+                });
             }
         }
-        this.$content.replaceWith($content);
-        this.$content = $content;
+        if (query) {
```

Comment:
Reduce diff by doing an `if (!query) return;` instead of nesting the whole unchanged code ?
Also why does the nested block have double indentation ?